backend/src/mirrors_qa_backend/routes/dependencies.py

```diff
@@ -0,0 +1,57 @@
+from typing import Annotated
+
+import jwt
+from fastapi import Depends, Header
+from jwt import exceptions as jwt_exceptions
+from pydantic import ValidationError as PydanticValidationError
+from sqlalchemy.orm import Session
+
+from mirrors_qa_backend import db, schemas
+from mirrors_qa_backend.db import gen_dbsession, models
+from mirrors_qa_backend.routes import http_errors
+from mirrors_qa_backend.settings import Settings
+
+DbSession = Annotated[Session, Depends(gen_dbsession)]
+
+
+def verify_authorization_header(
+    authorization: Annotated[str | None, Header()] = None
+) -> schemas.JWTClaims | None:
+    if authorization is None:
+        return None
+
+    header_parts = authorization.split(" ")
+    if len(header_parts) != 2 or header_parts[0] != "Bearer":  # noqa
+        raise http_errors.UnauthorizedError
+
+    token = header_parts[1]
+    try:
+        claims = jwt.decode(token, Settings.JWT_SECRET, algorithms=["HS256"])
+    except jwt_exceptions.ExpiredSignatureError:
+        raise http_errors.UnauthorizedError("Token has expired.") from None
+    except (jwt_exceptions.InvalidTokenError, jwt_exceptions.PyJWTError):
+        raise http_errors.UnauthorizedError from None
+
+    try:
+        claims = schemas.JWTClaims(**claims)
+    except PydanticValidationError:
+        raise http_errors.UnauthorizedError from None
+    return claims
+
+
+def get_current_worker(
+    session: DbSession,
+    claims: Annotated[schemas.JWTClaims | None, Depends(verify_authorization_header)],
+) -> models.Worker:
+    if claims is None:
```

Comment:
Because the global dependency ensures the header is set correctly, it can return a None if no header was set or the actual claims if everything was okay. Routes like this that need it to be an actual value respond with an error. I suppose I can remove it and put everything in the place it is used.